enterprise/ssl-impl/src/main/java/io/crate/protocols/ssl/SslContextProviderImpl.java

```diff
@@ -19,25 +19,38 @@
 package io.crate.protocols.ssl;
 
 import io.netty.handler.ssl.SslContext;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
 import org.elasticsearch.common.settings.Settings;
 
+@Singleton
 public class SslContextProviderImpl implements SslContextProvider {
 
     private volatile SslContext sslContext;
 
-    // required by service provider
-    public SslContextProviderImpl() {
+    private final Settings settings;
+
+    @Inject
+    public SslContextProviderImpl(Settings settings) {
+        this.settings = settings;
     }
 
     @Override
-    public SslContext getSslContext(Settings settings) {
+    public SslContext getSslContext() {
         if (sslContext == null) {
```

Comment:
Could assign the `sslContext` to a local variable to avoid the duplicate volatile read